### PR TITLE
episode selection menu

### PR DESCRIPTION
**Pull request by sandikbarr**
This PR satisfies requirements in #330, #331, and #345 to create an episode selection menu with search and multiselect capabilities. The episode selection menu doesn't yet do anything in the app until #324, so the episode data is loaded and processed into a store but the component is not shown anywhere.

### CODE REVIEW

File: src/app/shared/episode-select/episode-select-dropdown.component.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          scrollTop + clientHeight >= (scrollHeight - 100) &&
+          this.lastPage + 1 <= this.maxPages) {
+        this.loadEpisodes(this.lastPage + 1, this.searchTerm);
+        this.store.dispatch(new ACTIONS.GoogleAnalyticsEventAction({gaAction: 'episode-select-page-load', value: this.lastPage + 1}));
+      }
+    });
+  }
+
+  loadEpisodesOnSearch(search: string) {
+    this.loadEpisodes(1, search);
```
Comment (pstjean) [inline src/app/shared/episode-select/episode-select-dropdown.component.ts]: Does dispatching 2 separate actions make the most sense here? My gut says to roll them up into 1 but when I think about it I can't see any downside to doing it this way.

File: src/app/ngrx/reducers/selectors/error.selectors.ts [hunk 2]
```diff
@@ -39,7 +39,8 @@ export const select500ErrorReloadActions =
     }
     if (routerParams.metricsType === METRICSTYPE_DOWNLOADS) {
       if (episodeError && episodeError.status === 500) {
-        actions.push(new ACTIONS.CastleEpisodePageLoadAction({podcastId: routerParams.podcastId, page: 1, all: true}));
+        actions.push(new ACTIONS.CastleEpisodePageLoadAction({
```
[the remark this answers is not in the document]
Comment (sandikbarr) [reply src/app/ngrx/reducers/selectors/error.selectors.ts]: Yes! Most often hits a different instance and returns the data on the retry. We were having a lot of issues with out of memory on Castle, so this became kinda useful at one point.

File: src/app/shared/episode-select/episode-select-dropdown.component.ts (hunk 1 above)
Comment (sandikbarr) [reply src/app/shared/episode-select/episode-select-dropdown.component.ts]: I suppose the google analytics could be handled as a (side) effect when CastleEpisodeSelectPageLoadAction is dispatched, have an effect to dispatch the GoogleAnalyticsEventAction (which is itself an effect actually, `fromGAEvent$`) but it feels more direct to me to just dispatch both actions at their source. Mike Ryan did a great talk at ng-conf last year about designing Actions if you wanna learn more about their thoughts: https://www.youtube.com/watch?v=JmnsEvoy-gY

Comment (pstjean) [reply src/app/shared/episode-select/episode-select-dropdown.component.ts]: Yeah, thinking about it more I think that's correct. Moving it to the side effect level means that it's a layer of abstraction away from the user interaction, which is really what you're tracking. I'll check out that talk.

**Pull request merged**